### PR TITLE
Increase coverage

### PR DESCRIPTION
**Pull request by nkly**
Add several additional test cases. They increase statement/branch coverage from 85.6%/75% to 86.9%/83.3%

### CODE REVIEW

File: src/test/scala/com/advancedtelematic/campaigner/daemon/DeviceEventListenerSpec.scala [hunk 1]
```diff
@@ -30,6 +32,12 @@ class DeviceEventListenerSpec extends CampaignerSpec with DatabaseSpec with Devi
     Messages.DeviceEventMessage(campaign.namespace, event)
   }
 
+  private def genEventOfType(eventType: EventType): DeviceEventMessage = {
```
Comment (simao) [inline src/test/scala/com/advancedtelematic/campaigner/daemon/DeviceEventListenerSpec.scala]: if we are using Generators and not returning a `Gen[_]` then the method name should not start with `gen`

File: src/test/scala/com/advancedtelematic/campaigner/daemon/DeviceEventListenerSpec.scala [hunk 2]
```diff
@@ -65,4 +73,14 @@ class DeviceEventListenerSpec extends CampaignerSpec with DatabaseSpec with Devi
 
     deviceUpdateRepo.findByCampaign(campaign.id, DeviceStatus.failed).futureValue shouldBe Set(device)
   }
+
+  it should "raise IllegalArgumentException if the message is of non-acceptable type" in {
+    val msg = genEventOfType(EventType("campaign_accepted", 42))
+    listener.apply(msg).failed.futureValue.isInstanceOf[IllegalArgumentException]
```
Comment (simao) [inline src/test/scala/com/advancedtelematic/campaigner/daemon/DeviceEventListenerSpec.scala]: `listener.apply(msg).failed.futureValue shouldBe a[IllegalArgumentException]`

File: src/test/scala/com/advancedtelematic/campaigner/db/CampaignStatusRecalculateSpec.scala [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    val groups = GroupWithDevices.genNonEmptyListOf.generate
+
+    for {
+      campaign <- createCampaignWithoutStatus(maybeGroups = Some(groups.map(_.id)))
+      _ <- insertGroupStatsFor(campaign.id, groups.toList)
+      _ <- groupStatsRepo.updateGroupStatsAction(
+        campaign.id, groups.head.id, GroupStatus.cancelled, Stats(0, 0)).run
+      _ <- new CampaignStatusRecalculate().run
+      updatedCampaign <- campaignRepo.find(campaign.id)
+    } yield {
```
Comment (simao) [inline src/test/scala/com/advancedtelematic/campaigner/db/CampaignStatusRecalculateSpec.scala]: curly braces not needed for a single statement

Comment (simao) [reply src/test/scala/com/advancedtelematic/campaigner/db/CampaignStatusRecalculateSpec.scala]: Same in all examples

File: src/test/scala/com/advancedtelematic/campaigner/db/CampaignStatusRecalculateSpec.scala [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+import org.scalacheck.Gen
+import org.scalatest._
+import org.scalatest.concurrent.ScalaFutures
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import slick.jdbc.MySQLProfile.api._
+
+final class CampaignStatusRecalculateSpec
+    extends TestKit(ActorSystem("CampaignStatusRecalculateSpec"))
+    with AsyncFlatSpecLike
```
Comment (simao) [inline src/test/scala/com/advancedtelematic/campaigner/db/CampaignStatusRecalculateSpec.scala]: If you really like this style, keep it, but this leads to rather weird testing, for comprehensions everywhere, beating non future values into futures, and always having to end a future with an assertion...

File: src/test/scala/com/advancedtelematic/campaigner/daemon/DeviceEventListenerSpec.scala (hunk 2 above)
Comment (nkly) [reply src/test/scala/com/advancedtelematic/campaigner/daemon/DeviceEventListenerSpec.scala]: Good one :+1:

File: src/test/scala/com/advancedtelematic/campaigner/db/CampaignStatusRecalculateSpec.scala (hunk 3 above)
Comment (nkly) [reply src/test/scala/com/advancedtelematic/campaigner/db/CampaignStatusRecalculateSpec.scala]: Sure. Do you follow a particular code style in projects? Where can I take a look at it? Would be nice to know the rules before playing the game :)

File: src/test/scala/com/advancedtelematic/campaigner/db/CampaignStatusRecalculateSpec.scala (hunk 4 above)
Comment (nkly) [reply src/test/scala/com/advancedtelematic/campaigner/db/CampaignStatusRecalculateSpec.scala]: TBH I just spotted `AsyncFlatSpec` usage in one of the tests and just followed it without much research. Later I've discovered `ScalaFutures` mixin which looks much better IMO. I'll refactor this test.

File: src/test/scala/com/advancedtelematic/campaigner/daemon/DeviceEventListenerSpec.scala (hunk 1 above)
Comment (nkly) [reply src/test/scala/com/advancedtelematic/campaigner/daemon/DeviceEventListenerSpec.scala]: Sure. Same question about code style as in a comment about `yield` braces applies.

File: src/test/scala/com/advancedtelematic/campaigner/db/CampaignStatusRecalculateSpec.scala (hunk 3 above)
Comment (simao) [reply src/test/scala/com/advancedtelematic/campaigner/db/CampaignStatusRecalculateSpec.scala]: We did at some point use scalastyle.xml but it caused more discussion than the time to discuss the changes ;) So we don't follow any particular code style except for the projects that do include scalastyle.xml

Comment (nkly) [reply src/test/scala/com/advancedtelematic/campaigner/db/CampaignStatusRecalculateSpec.scala]: Okay, got it. Thanks!

**Pull request merged**